features/step_definitions/user.rb

```diff
@@ -72,3 +72,51 @@
 Then(/^(.*) should not be able to sign up$/) do |username|
   expect(user_named(username).persisted?).to be_falsey
 end
+
+
+When(/^(.+) visits the password recovery page$/) do |username|
+  visit new_user_password_path
+end
+
+And(/^(.+) inserts his email address$/) do |username|
+  user = user_named(username)
+  fill_in :user_email, with: user.email
+end
+
+And(/^submits$/) do
+  click_button I18n.t('devise.registrations.reset_password')
+end
+
+Then(/^(\w+) gets an Email with a recovery link$/) do |username|
+  user = user_named(username)
+  reset_password_mail = ActionMailer::Base.deliveries.last
+  expect(reset_password_mail.subject).to have_content('asswor') # (P|p)asswor(d|t)
+  expect(reset_password_mail.to[0]).to eq(user.email)
+  expect(reset_password_mail.body.to_s).to have_css('a')
```

Comment:
Maybe check if the url of the link is correct.

Reply:
how?

Reply:
Check for the token somewhere in the link?